client/question/services/ConversationManagerServiceSpec.js

```diff
@@ -411,54 +435,54 @@ describe('ConversationManagerService', function() {
       verifySubmissions([], done, [{
         code: codeThatFailsBuggyOutputForTaskOne,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          'Mock BuggyOutputTest Message One for task1')
+          ['Mock BuggyOutputTest Message One for task1'])
       }]);
     });
 
     it('should show task 1 f/b over task 2 buggy-output f/b', function(done) {
       verifySubmissions([], done, [{
         code: codeThatFailsBuggyOutputForTaskTwo,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          null, "Input: \"task1suite1test1\"")
+          [null, "Input: \"task1suite1test1\""])
       }]);
     });
 
     it('should show a new message only if the code changes', function(done) {
       verifySubmissions([], done, [{
         code: codeThatFailsBuggyOutputForTaskOne,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          'Mock BuggyOutputTest Message One for task1')
+          ['Mock BuggyOutputTest Message One for task1'])
       }, {
         // The code has not changed, so the message stays the same.
         code: codeThatFailsBuggyOutputForTaskOne,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          'Mock BuggyOutputTest Message One for task1')
+          ['Mock BuggyOutputTest Message One for task1'])
       }, {
         // The code has changed, so the message changes.
         code: anotherCodeThatFailsBuggyOutputForTaskOne,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          'Mock BuggyOutputTest Message Two for task1')
+          ['Mock BuggyOutputTest Message Two for task1'])
       }]);
     });
 
     it('shows correctness f/b after all buggy hints are shown', function(done) {
       verifySubmissions([], done, [{
         code: codeThatFailsBuggyOutputForTaskOne,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          'Mock BuggyOutputTest Message One for task1')
+          ['Mock BuggyOutputTest Message One for task1'])
       }, {
         code: anotherCodeThatFailsBuggyOutputForTaskOne,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          'Mock BuggyOutputTest Message Two for task1')
+          ['Mock BuggyOutputTest Message Two for task1'])
       }, {
         code: codeThatFailsBuggyOutputForTaskOne,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          'Mock BuggyOutputTest Message Three for task1')
+          ['Mock BuggyOutputTest Message Three for task1'])
       }, {
         // At this point, we have run out of buggy-output test feedback.
         code: anotherCodeThatFailsBuggyOutputForTaskOne,
         expectedFeedback: ExpectedFeedbackObjectFactory.create(
-          null, null, '', false, function(submissionResult) {
+          null, '', false, function(submissionResult) {
```

Comment:
If we're going to have the first arg be all the feedback, then the first arg should be an empty list if no feedback is expected (not null).